File: public/docs/_examples/component-styles/ts/app/hero-controls.component.ts
import {Component, Input} from 'angular2/core';
import {Hero} from './hero';

// #docregion inlinestyles
@Component({
  selector: 'hero-controls',
  template: `
    <style>
      button {
        background-color: white;
        border: 1px solid #777;
      }
    </style>
    <h3>Controls</h3>
    <button (click)="activate()">Activate</button>
  `
})
export class HeroControlsComponent {
// #enddocregion inlinestyles

  @Input() hero:Hero;

  activate() {
    this.hero.active = true;
  }
}

File: public/docs/_examples/component-styles/dart/lib/hero_controls_component.dart
import 'package:angular2/core.dart';
import 'hero.dart';

// #docregion inlinestyles
@Component(
    selector: 'hero-controls',
    template: '''
      <style>
        button {
          background-color: white;
          border: 1px solid #777;
        }
      </style>
      <h3>Controls</h3>
      <button (click)="activate()">Activate</button>''')
class HeroControlsComponent {
  @Input()
  Hero hero;

  void activate() {
    hero.active = true;
  }
}
